src/bundle/Resources/public/js/scripts/admin.anchor.navigation.js

```diff
@@ -0,0 +1,65 @@
+(function(global, doc, eZ) {
+    const sectionsPosition = {};
+    const scrollOffset = 400;
+    const formContainerNode = doc.querySelector('.ibexa-content-edit-content');
+    const isVerticalScrollVisible = () => {
+        const { scrollHeight, offsetHeight } = formContainerNode;
+
+        return scrollHeight > offsetHeight;
+    };
+    const showSection = (sectionId) => {
+        doc.querySelectorAll('.ibexa-anchor-navigation-menu__btn').forEach((btn) => {
+            const { anchorTargetSectionId } = btn.dataset;
+
+            btn.classList.toggle('ibexa-anchor-navigation-menu__btn--active', anchorTargetSectionId === sectionId);
+        });
+
+        doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
+            const { anchorSectionId } = section.dataset;
+
+            section.classList.toggle('ibexa-anchor-navigation-sections__section--active', anchorSectionId === sectionId);
+        });
+    };
+    const navigateTo = (event) => {
+        let startSection = 0;
+        let endSection = 0;
+        const { anchorTargetSectionId } = event.currentTarget.dataset;
+
+        doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
```

Comment:
what would you say for sth like
```
let prevSection = 0;
doc.querySelectorAll('.ibexa-anchor-navigation-sections__section').forEach((section) => {
    const { anchorSectionId } = section.dataset;

    sectionsPosition[anchorSectionId] = prevSection;
    prevSection = section.offsetHeight;
});
```
as far as I understand you dont need both variables

Reply:
I am not sure what you are proposing, @GrabowskiM. You only changed `startSection` name to `prevSection` and position of its declaration? Maybe I missed something in your code excerpt.

I like change of the variable name, but I wonder, do we even need `sectionsPosition`, @lucasOsti?
Because it is used only in 1 place AFAIK:
https://github.com/ezsystems/ezplatform-admin-ui/pull/1770/files#diff-2b5ffe8475c06a6575af7905596bab7de43658bb4bd13a9b658da19d8d5de2b1R36
`top: sectionsPosition[anchorTargetSectionId],`

My proposition in other comment.